agent/agent.go

```diff
@@ -1407,5 +1407,12 @@ func expandDirectory(dir string) (string, error) {
 		}
 		dir = filepath.Join(home, dir[1:])
 	}
+	if !filepath.IsAbs(dir) {
```

Comment:
Since this is done before `os.ExpandEnv`, it will incorrectly think the path isn’t absolute when the value is e.g. `$HOME`. Moving this too after that should fix the test that’s failing.

Reply:
Thanks for the suggestion but $HOME not being used as an absolute path